drivers/sensor/bosch/bmp180/bmp180.c

```diff
@@ -0,0 +1,485 @@
+		return -EBUSY;
+	}
+#endif /* CONFIG_PM_DEVICE */
+
+	pm_device_busy_set(dev);
+
+	ret = read_raw_temperature(dev);
+	if (ret < 0) {
+		goto error;
+	}
+
+	ret = read_raw_pressure(dev);
+
+error:
+	pm_device_busy_clear(dev);
+	return ret;
+}
+
+static void bmp180_compensate_temp(struct bmp180_data *data)
+{
+	int32_t partial_data1;
+	int32_t partial_data2;
+	struct bmp180_cal_data *cal = &data->cal;
+
+	partial_data1 = (data->sample.raw_temp - cal->ac6) * cal->ac5 / 0x8000;
+	partial_data2 = cal->mc * 0x800 / (partial_data1 + cal->md);
+
+	/* Store for pressure calculation */
+	data->sample.comp_temp = (partial_data1 + partial_data2);
+}
+
+static int bmp180_temp_channel_get(const struct device *dev,
+				   struct sensor_value *val)
+{
+	struct bmp180_data *data = dev->data;
+
+	if (data->sample.comp_temp == 0) {
+		bmp180_compensate_temp(data);
+	}
+
+	float ftmp = ((data->sample.comp_temp+8)>>4) / 10.0;
+	int64_t tmp = floor(ftmp * 1000000);
+
+	val->val1 = tmp / 1000000;
+	val->val2 = tmp % 1000000;
+
+	return 0;
+}
+
+static uint32_t bmp180_compensate_press(struct bmp180_data *data)
+{
+	uint64_t partial_B7;
+	uint64_t partial_B4;
+	int32_t partial_X1;
+	int32_t partial_X2;
+	int32_t partial_X3;
+	int32_t partial_B3;
+	int32_t partial_B6;
+	uint32_t comp_press;
+	int32_t raw_pressure = data->sample.raw_press;
+	struct bmp180_cal_data *cal = &data->cal;
+
+	partial_B6 = data->sample.comp_temp - 4000;
+	partial_X1 = (cal->b2 * partial_B6 * partial_B6) / 0x800000;
+	partial_X2 = (cal->ac2 * partial_B6) / 0x800;
+	partial_X3 = partial_X1 + partial_X2;
+	partial_B3 = (((cal->ac1 * 4 + partial_X3)<<data->osr_pressure)+2) / 4;
+
+	partial_X1 = (cal->ac3 * partial_B6) / 0x2000;
+	partial_X2 = (cal->b1 * partial_B6 * partial_B6) / 0x10000000;
+	partial_X3 = (partial_X1 + partial_X2 + 2) / 4;
+	partial_B4 = (cal->ac4 * (partial_X3 + 32768)) / 0x8000;
+	partial_B7 = (raw_pressure - partial_B3) * (50000>>data->osr_pressure);
+
+	comp_press = (uint32_t)(partial_B7 / partial_B4 * 2);
+
+	partial_X1 = (comp_press>>8)*(comp_press>>8);
+	partial_X1 = (partial_X1 * 3038) / 0x10000;
+	partial_X2 = ((int32_t)(-7357 * comp_press))/ 0x10000;
+	comp_press += (partial_X1 + partial_X2 + 3791) / 16;
+
+	/* returned value is Pa */
+	return comp_press;
+}
+
+static int bmp180_press_channel_get(const struct device *dev,
+				    struct sensor_value *val)
+{
+	struct bmp180_data *data = dev->data;
+
+	if (data->sample.comp_temp == 0) {
+		bmp180_compensate_temp(data);
+	}
+
+	uint32_t tmp = bmp180_compensate_press(data);
+
+	/* tmp is in hundredths of Pa. Convert to kPa as specified in sensor
+	 * interface.
+	 */
+	val->val1 = tmp / 100000;
+	val->val2 = (tmp % 100000) * 10;
+
+	return 0;
+}
+
+static int bmp180_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_PRESS:
+		bmp180_press_channel_get(dev, val);
+		break;
+
+	case SENSOR_CHAN_DIE_TEMP:
```

Comment:
I owe you a reply on this,
if the sensor is mounted on a pcb where the board emit heat to the sensor, this is a DIE_TEMP because its differs from the ambient temperature. But if the senor sit on a part which is isn't influenced from self heating we measure the environment.  Depending on this both channel apply.

Reply:
I do not know if any other temperature sensor does this. Also if you say its dependent on where it is mounted - it will be kind of strange to have this if it is not case, no?

Reply:
Resolved, only use DIE_TEMP because the bmp180 is a digital pressure sensor not a temperature sensor.